linkerd/stack/src/make_thunk.rs

```diff
@@ -0,0 +1,53 @@
+use futures::future;
+use linkerd2_error::Never;
+use std::task::{Context, Poll};
+
+#[derive(Clone, Debug)]
+pub struct MakeThunk<M> {
+    make: M,
+}
```

Comment:
it might be nice to have a comment explaining what this thing does exactly, for future readers?

i agree that `MakeThunk` describes what it does in a general sense, but it might be worth naming it something that uses `tower`/linkerd-specific language? e.g., when i wrote a version of this for the control plane connector (which we no longer use), i initially named it `Curried`, and then thought that seemed too abstract, and called it `WithTarget`.

sorry for bikeshedding, it fundamentally doesn't matter.

Reply:
I documented it. I don't think `WithTarget` is all that clear, personally (especially when compared with `MapTarget` -- I'd expect `WithTarget` to take a constant target to be reused for all requests, or something...). Anyway, we can easily change this later.